### PR TITLE
Add fields to reservation statistics

### PR DESCRIPTION
**Pull request by matti-lamppu**
## 🛠️ Changelog
[//]: # "Describe the changes in this pull request here."

- Adds fields as described in the JIRA ticket
- Refactors and groups statistics model fields

## 🧪 Test plan
[//]: # "Help your fellow reviewer and write a short description of the fastest way to test your changes."

- Automated tests

## 🚧 Dependencies
[//]: # "Is this PR dependent on other changes outside this repository? Describe the changes, or add links to them."
[//]: # "e.g. This PR breaks X and is waiting for frontend changes before merging."

- None

## 🎫 Tickets
[//]: # "This pull request resolves all or part of the following ticket(s)."

- [TILA-3424][TILA]

[TILA]: https://helsinkisolutionoffice.atlassian.net/browse/TILA-3424

[TILA-3424]: https://helsinkisolutionoffice.atlassian.net/browse/TILA-3424?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: reservations/statistic_utils.py [hunk 1]
```diff
@@ -1,52 +1,62 @@
+from reservations.choices import CustomerTypeChoice
 from reservations.models import Reservation, ReservationStatistic, ReservationStatisticsReservationUnit
 
 
 def create_or_update_reservation_statistics(reservation_pk: Reservation) -> None:
     reservation = Reservation.objects.get(pk=reservation_pk)
     recurring = getattr(reservation, "recurring_reservation", None)
+    requires_org_info = (
```
Comment (vergama) [inline reservations/statistic_utils.py:8]: organisation name is still wanted if the customer type is non-individual but unreg.association

Comment (matti-lamppu) [reply reservations/statistic_utils.py:8]: Wait, isn't that what this already does? Or did you mean:

```python
requires_org_name = reservation.reservee_type != CustomerTypeChoice.INDIVIDUAL
requires_org_id = reservation.reservee_is_unregistered_association and requires_org_name
```

Comment (vergama) [reply reservations/statistic_utils.py]: Yes, that.

Comment (matti-lamppu) [reply reservations/statistic_utils.py]: OK, it was missing a "not" before, but I added this so it should be fixed now.

**Review by vergama — APPROVED** [review]

File: tilavarauspalvelu/settings.py [hunk 2]
```diff
@@ -797,6 +798,8 @@ def SEARCH_SETTINGS(cls):
     # Turn off reservation unit hierarchy updates from signals during tests,
     # since they slow them down a lot in CI. Refresh should be called manually when needed.
     UPDATE_RESERVATION_UNIT_HIERARCHY = False
+    # Turn off statistics saving during tests for performance reasons
+    SAVE_RESERVATION_STATISTICS = False
```
Comment (ranta) [inline tilavarauspalvelu/settings.py:802]: ❤️

**Review by ranta — APPROVED** [review]

**Comment by matti-lamppu** [comment]
There might be a non-null constraint issue here, I should take a look before merging.

**Pull request closed without merging**